docs/content/sips/xxx-manifest-redesign.md

```diff
@@ -0,0 +1,123 @@
+title = "SIP xxx - Application Manifest Redesign"
+template = "main"
+date = "2022-05-20T13:22:30Z"
+---
+
+Summary: A new design for Application Manifests (`spin.toml`)
+
+Owner: lann.martin@fermyon.com
+
+Created: May 20, 2022
+
+## Background
+
+Spin's Application Manifest file (usually named `spin.toml`) describes a Spin application's metadata,
+components, and trigger configurations. The current design supports a single trigger type and combines
+component and trigger configuration with an implicit one-to-one relationship:
+
+```toml
+name = "hello-world"
+trigger = { type = "http", base = "/hello" }
+
+[[component]]
+id = "world"
+# ...
+[component.trigger]
+route = "/world"
+```
+
+In order to support future features, we would like to enable more flexible trigger configurations:
+
+- Multiple trigger types in an application
+- Multiple triggers associated with a single component
+- One trigger associated with multiple components
+
+## Proposal
+
+### Global trigger config
+
+In order to allow for multiple trigger types, we remove the top-level `trigger.type` field.
+We also move all application metadata from the top-level config to a new `[application]`
+section, and all "global trigger config" to new `[application.trigger.<type>]` sections:
+
+```toml
+[application]
+name = "hello-world"
+[application.trigger.http]
```

Comment:
This is a table rather than a table array, which means that (at some point) we can have multiple triggers of different types, but not multiple triggers of the same type, is that right?  Or do we envisage revving the version at that point?

(Of course I might just be misunderstanding my TOML syntax here!)

Reply:
This is "application-wide HTTP trigger config". I'm not entirely happy with the syntax, but didn't come up with anything better.